python/ql/src/experimental/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -523,6 +597,30 @@ predicate comprehensionStoreStep(CfgNode nodeFrom, Content c, CfgNode nodeTo) {
   c instanceof ListElementContent
 }
 
+/**
+ * In
+ * ```python
+ * obj.foo = x
+ * ```
+ * data flows from `x` to (the post-update node for) `obj` via assignment to `foo`.
+ */
+predicate attributeStoreStep(CfgNode nodeFrom, Content c, PostUpdateNode nodeTo) {
+  exists(AssignStmt a, Attribute attr |
```

Comment:
I'm a bit puzzled by the choice of modelling this at the AST layer rather than the CFG layer. What if splitting results in `a.getValue().getAFlowNode()` having multiple values? Won't this potentially lead to cross-talk between the split branches?

Also, `AssignStmt` is rather specific, here. We'll be excluding augmented assignments, annotated assignments, etc. and that doesn't seem right to me.

Reply:
I think you are right. This was me getting to grips with the API in the early days. I found `DefinitionNode` which solves the problem neatly.

Reply:
I also managed to write `attributeReadStep` in a nicer way :-)